atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -41,7 +41,13 @@ def run(self):
         """
         pull base image
         """
-        base_image = self.workflow.builder.base_image
+        build_json = get_build_json()
+        spec = build_json.get("spec")
+        try:
+            image_id = spec['triggeredBy'][0]['imageChangeBuild']['imageID']
+            base_image = ImageName.parse(image_id)
+        except:
```

Comment:
Better to be explicit about what we want to catch here, otherwise  a typo might always send us into the except block. Looks like we want TypeError (in case `spec` is `None`), KeyError (for the dict lookups), and IndexError (for the list lookup).

If we cannot parse the imageID provided in triggeredBy, that is probably a bug in `ImageName.parse()` (or in OpenShift) so it should probably be fatal rather than falling back to what the Dockerfile says the parent image is.

Reply:
Fixed I think.
I'm not sure if the second paragraph means I should add nested trys to specifically blow up if the parse fails.